Fix constructor in MethodPluginScan

fedebetti opened the pull request:
Fixes https://github.com/gammacombo/gammacombo/issues/280
This seems due to a wrong order in a MethodPluginScan constructor:
```
MethodPluginScan::MethodPluginScan(MethodProbScan* s, PDF_Datasets* pdf, const OptParser* opt) : MethodAbsScan(opt) {
  constructorHelper(s);
  obsName = pdf->getObsName();
  w = pdf->getWorkspace();
};
```
Moving `constructorHelper(s);` after `obsName` and `w` definition fixes it.

fedebetti committed:
fix constructor in MethodPluginScan

fedebetti commented:
@tmombaecher this is what we discussed this afternoon.
@tpajero FYI

tpajero reviewed (approved):
Looks good to me. I am not familiar with actions on GitHub, maybe @matthewkenzie knows better

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/pre-commit/pre-commit-hooks: v4.6.0 → v6.0.0](https://github.com/pre-commit/pre-commit-hooks/compare/v4.6.0...v6.0.0)
- [github.com/pre-commit/mirrors-clang-format: v16.0.6 → v21.1.8](https://github.com/pre-commit/mirrors-clang-format/compare/v16.0.6...v21.1.8)
- [github.com/astral-sh/ruff-pre-commit: v0.12.0 → v0.14.10](https://github.com/astral-sh/ruff-pre-commit/compare/v0.12.0...v0.14.10)

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

tmombaecher commented:
@fedebetti can you try to make an empty commit to try again? So we finally close this... It may just have been stuck once.

git commit --allow-empty -m "retrigger CI"
git push

If that still doesn't help, it can be that the workflow is not properly configured for PR's from forks. Then we need to check again.

fedebetti commented:
> @fedebetti can you try to make an empty commit to try again? So we finally close this... It may just have been stuck once.
> 
> git commit --allow-empty -m "retrigger CI" git push
> 
> If that still doesn't help, it can be that the workflow is not properly configured for PR's from forks. Then we need to check again.

Done

tpajero commented:
No idea of what is the problem with the configuration of the GitHub actions

I've made the changes in a new Pull request https://github.com/gammacombo/gammacombo/pull/286

Many thanks @fedebetti!
